File: android/src/main/java/com/thouland/plugins/audioplayer/AudioplayerPlugin.java
package com.thouland.plugins.audioplayer;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Audioplayer")
public class AudioplayerPlugin extends Plugin implements onCompletedListener {

    private Audioplayer implementation = new Audioplayer();

    @PluginMethod
    public String start(PluginCall call) {

        String value = call.getString("value");
        JSObject ret = new JSObject();
        implementation.setup(this);
        ret.put("value", implementation.start(call,value));
        call.resolve(ret);

        return "yes";
    }


    @PluginMethod
    public String pause(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", implementation.pause());
        call.resolve(ret);
        return "yes";
    }

    @PluginMethod
    public String stop(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", implementation.stop());
        call.resolve(ret);
        return "yes";
    }



    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @Override
    public void onCompleted() {
        notifyListeners("nextSong", null);
    }
}

File: ios/Plugin/AudioplayerPlugin.swift
import Foundation
import Capacitor
import AVFoundation

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AudioplayerPlugin)
public class AudioplayerPlugin: CAPPlugin {
     private var implementation :Audioplayer?

    
    @objc func start(_ call: CAPPluginCall)  -> String {
        let value = call.getString("value") ?? ""
        var playItem = AVPlayerItem(url: URL(string: value)!)
        var player = AVPlayer(playerItem: playItem)
        implementation = Audioplayer(player: player, despPlugin: self)
        implementation?.setup(value: self)
        call.resolve([
            "value": implementation?.start(value: value)
        ])
        return "yes"
    }
    @objc func onCompleted() {
                notifyListeners("nextSong", data: nil);
           }

        @objc func pause(_ call: CAPPluginCall)  -> String {
                call.resolve([
                    "value": implementation?.pause()
                ])
              return "yes"
    }
      
    @objc func stop(_ call: CAPPluginCall)  -> String {
                call.resolve([
                    "value": implementation?.stop()
                ])
          return "yes"
}
        @objc func echo(_ call: CAPPluginCall)  -> String {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation?.echo(value)
        ])
            return "yes"
    }

}